packages/scaffold-ui/src/views/w3m-switch-active-chain-view/index.ts

```diff
@@ -70,28 +70,32 @@ export class W3mSwitchActiveChainView extends LitElement {
             Connected wallet doesn't support connecting to ${switchedChainNameString} chain. You
             need to connect with a different wallet.
           </wui-text>
-          <wui-button size="md" @click=${this.switchActiveChain.bind(this)}>Switch</wui-button>
+          <wui-button
+            data-testid="w3m-switch-active-chain-button"
+            size="md"
+            @click=${this.switchActiveChain.bind(this)}
+            >Switch</wui-button
+          >
         </wui-flex>
       </wui-flex>
     `
   }
 
   // -- Private Methods ------------------------------------ //
-  private async switchActiveChain() {
+  private switchActiveChain() {
     if (!this.switchToChain) {
       return
     }
 
+    ChainController.setIsSwitchingNamespace(true)
+
     if (this.caipNetwork) {
-      await ChainController.switchActiveNetwork(this.caipNetwork)
+      ChainController.setActiveCaipNetwork(this.caipNetwork)
     } else {
       ChainController.setActiveNamespace(this.switchToChain)
     }
```

Comment:
should we encapsulate this in ChainController directly? I think `setActiveCaipNetwork`  should set the isSwitchingNamespace flag directly

Reply:
Totally, and it already has that call inside, removing here